onnx/defs/schema.h

```diff
@@ -25,6 +25,40 @@
 #include "onnx/onnx-operators_pb.h"
 namespace ONNX_NAMESPACE {
 
+struct FunctionBodyQueryContext {
+  virtual const AttributeProto* getAttribute(const std::string& name) const = 0;
+  virtual size_t getNumInputs() const = 0;
```

Comment:
do you allow that an optional input exists but the other optional input before it does not exist?

Reply:
Yes. with @gramalingam and @TMVector 's comment implemented, the functionProto will keep the amount of inputs (required and optional). Thus optional input may be in any combination.

Reply:
I think @linkerzhang may be suggesting that we enrich this interface by explicitly adding interface method like:
```
   virtual bool hasInput(size_t index);
```
I think that is useful because intermediate optional inputs may be absent. Similarly, I think we will need
```
  virtual const TypeProto* getInputType(size_t index) const = 0;
```
to allow type-dependent specialization.

(However, I am okay if we add the above as a separate PR also, to get this PR in first.)

Reply:
YES, Rama is correct!

If allowing the case that I mentioned, then current interface is not enough.